### DISCUSSION TITLE
What is the difference on android of isLanguageAvailable and isLanguageInstalled???

### DISCUSSION THREAD

**Discussion by echogit** [dlutton/flutter_tts#451] [Q&A]
I created an app for children that supports many languages, and the user can switch to another language to learn it.
In all android devices I tested in Brazil, it had from default support for spanish, portuguese (Brazil and Portugal), English (American and British).
I added a verification if the language is not available, it should appear an icon next to the flag of the language. But when a friend of mine tested, it didn't show the icon, meaning that the isLanguageAvailable returned true, even if didn't have the language, but the TTS didn't work for that language. After explicitly downloading it it worked. Should I use isLanguageInstalled instead?

**Comment by dlutton** [comment]
isLanguageInstalled is checking for KEY_FEATURE_NOT_INSTALLED: 

https://developer.android.com/reference/android/speech/tts/TextToSpeech.Engine#KEY_FEATURE_NOT_INSTALLED 

isLanguageAvailable is checking or isLanguageAvailable:

https://developer.android.com/reference/android/speech/tts/TextToSpeech#isLanguageAvailable(java.util.Locale)

**Comment by echogit** [comment]
From what I read in this documentation you mentioned, if language isn't installed, running setLanguage would trigger the download, correct?
In the case I mentioned, it just didn't work, and the download had to be done explicitly on the google TTS configuration.

**Comment by echogit** [comment]
Can I use this:

```
   Map<String, bool> newAvailability = {};
    for (var language in appSupportedLanguages) {
      String locale =
          language['locale'] ?? ""; 
      bool isAvailable = await _flutterTts.isLanguageAvailable(locale);
      newAvailability[locale] = isAvailable;
      print("$locale available? $isAvailable");
    }

    if (isAndroid) {
      for (var language in appSupportedLanguages) {
        String locale =
            language['locale'] ?? ""; 
        bool isInstalled = await _flutterTts.isLanguageInstalled(locale);
        newAvailability[locale] = isInstalled;
        print("$locale installed? $isInstalled");
      }
    }
```
How could I start a process to download automatically all languages that are not yet downloaded?
Could I just call:
`await _flutterTts.setLanguage(locale)?`
For each language that is not installed?
